app/api/domains/osu.py

```diff
@@ -261,17 +261,25 @@ async def osuGetBeatmapInfo(
         grades = ["N", "N", "N", "N"]
 
         await db_conn.execute(
-            "SELECT grade, mode FROM scores_rx "
+            "SELECT grade, mode FROM scores "
             "WHERE map_md5 = :map_md5 AND userid = :user_id "
-            "AND status = 2",
-            {"map_md5": row["md5"], "user_id": player.id},
+            "AND mode = :mode AND status = 2",
+            {
+                "map_md5": row["md5"],
+                "user_id": player.id,
+                "mode": player.status.mode,
+            },
         )
 
         for score in await db_conn.fetch_all(
-            "SELECT grade, mode FROM scores_rx "
+            "SELECT grade, mode FROM scores "
             "WHERE map_md5 = :map_md5 AND userid = :user_id "
-            "AND status = 2",
-            {"map_md5": row["md5"], "user_id": player.id},
+            "AND mode = :mode AND status = 2",
+            {
+                "map_md5": row["md5"],
+                "user_id": player.id,
+                "mode": player.status.mode,
+            },
```

Comment:
i guess using mode for these is actually a bugfix

Reply:
not sure how it will work on the osu-side for rx/ap

Reply:
may want testing

Reply:
this is called by osu when it sees fit per mode, so it is a bug fix and also QoL fix. should be fine for rx/ap, it's just rng